api/android/api/src/androidTest/java/org/nnsuite/nnstreamer/APITestCommon.java

```diff
@@ -161,19 +163,84 @@ public static File getTFLiteAddModel() {
      * CPU: "custom=ModelFWType:CAFFE,ExecutionDataType:FLOAT32,ComputingUnit:CPU"
      * GPU: "custom=ModelFWType:CAFFE,ExecutionDataType:FLOAT32,ComputingUnit:GPU,GpuCacheSource:/sdcard/nnstreamer/"
      */
-    public static String getSNAPCaffeOption(boolean useGPU) {
+    public static String getSNAPCaffeOption(SNAPComputingUnit CUnit) {
         String option = "ModelFWType:CAFFE,ExecutionDataType:FLOAT32,InputFormat:NHWC,OutputFormat:NCHW,";
 
-        if (useGPU) {
-            String root = Environment.getExternalStorageDirectory().getAbsolutePath();
-            option = option + "ComputingUnit:GPU,GpuCacheSource:" + root + "/nnstreamer/";
-        } else {
-            option = option + "ComputingUnit:CPU";
+        switch (CUnit) {
+            case CPU:
+                option = option + "ComputingUnit:CPU";
+            case GPU:
+                String root = Environment.getExternalStorageDirectory().getAbsolutePath();
+                option = option + "ComputingUnit:GPU,GpuCacheSource:" + root + "/nnstreamer/";
+            case DSP:
+                break;
+            case NPU:
+                break;
         }
 
         return option;
     }
 
+    /**
+     * Gets the option string to run Tensorflow model for SNAP.
+     *
+     * CPU: "custom=ModelFWType:CAFFE,ExecutionDataType:FLOAT32,ComputingUnit:CPU"
+     * GPU: Not supported for Tensorflow model
+     * DSP: "custom=ModelFWType:CAFFE,ExecutionDataType:FLOAT32,ComputingUnit:DSP"
+     * NPU: "custom=ModelFWType:CAFFE,ExecutionDataType:FLOAT32,ComputingUnit:NPU"
+     */
+    public static String getSNAPTensorflowOption(SNAPComputingUnit CUnit) {
+        String option = "ModelFWType:TENSORFLOW,ExecutionDataType:FLOAT32,InputFormat:NHWC,OutputFormat:NHWC,";
+
+        switch (CUnit) {
+            case CPU:
+                option = option + "ComputingUnit:CPU";
+                break;
+            case GPU:
+                /* Not supported */
+                break;
+            case DSP:
+                option = option + "ComputingUnit:DSP";
+                break;
+            case NPU:
+                option = option + "ComputingUnit:NPU";
+                break;
```

Comment:
You may handle a default case here.
```suggestion
                break;
         case  GPU:
         case default:
                break;
```